### PR TITLE
ui: add graphs for clock offset mean and stddev

### PR DESCRIPTION
**Pull request by vivekmenezes**
This PR doesn't work. I'm obviously doing something wrong. I'll also point out that node heartbeat latency metrics on the "distributed" page are also not working.

Release note: None

### CODE REVIEW

**Comment by a-robinson** [comment]
Not working on what cluster? A local cluster?

The node heartbeat latency metrics are working on, for example, https://cyan.crdb.io:8080/#/cluster/all/distributed

**Comment by vivekmenezes** [comment]
@a-robinson yeah pointing this change at cyan does display the heartbeat latency metric. This change is still unable to display what it wants. I do see the variables on 

https://cyan.crdb.io:8080/_status/vars

File: pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx [hunk 1]
```diff
@@ -121,5 +121,18 @@ export default function (props: GraphDashboardProps) {
         }
       </Axis>
     </LineGraph>,
+
+    <LineGraph title="Mean Clock Offset" sources={nodeSources}>
+      <Axis label="nanos">
+        <Metric name="cr.node.clock.offset.meannanos" title="Mean Clock Offset" nonNegativeRate />
```
Comment (vilterp) [inline pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx]: looks like a typo in the metric name: it's `cr.node.clock-offset.meannanos`. https://github.com/cockroachdb/cockroach/blob/3dbe1dae8e734a27df53591f882efcd1543de9cd/pkg/rpc/clock_offset.go#L48

**Comment by couchand** [comment]
Could you also add a tooltip with a description of the measure we're charting?

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx, line 126 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21411#-L2zjso1YKHJE9389Qfa:-L2zjso1YKHJE9389Qfb:b-59o1xk) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b1d817d7ea3217e4b95592d2a76046ee5c47b04/pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx#L126)):*
> ```XML
> 
>     <LineGraph title="Mean Clock Offset" sources={nodeSources}>
>       <Axis label="nanos">
> ```

Should this be `units={AxisUnits.Duration}`?

---

*[pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx, line 131 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21411#-L2zjumTs6KArPEMSrmA:-L2zjumTs6KArPEMSrmB:btht497) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b1d817d7ea3217e4b95592d2a76046ee5c47b04/pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx#L131)):*
> ```XML
>     </LineGraph>,
> 
>     <LineGraph title="StdDev Clock Offset" sources={nodeSources}>
> ```

Would it makes sense to combine these into a single chart?

---

*[pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx, line 133 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21411#-L2zjx15rOYrIhPBZ89x:-L2zjx15rOYrIhPBZ89y:bln8min) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b1d817d7ea3217e4b95592d2a76046ee5c47b04/pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx#L133)):*
> ```XML
>     <LineGraph title="StdDev Clock Offset" sources={nodeSources}>
>       <Axis label="nanos">
>         <Metric name="cr.node.clock.offset.stddevnanos" title="Std Dev Clock Offset" nonNegativeRate />
> ```

I would flip around the titles for both series, and spell this one out in full: "Clock Offset Mean" and "Clock Offset Standard Deviation"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21411#-:-L2zk19JKYvwwDqp-JKS:bajqdqz)*

**Comment by vivekmenezes** [comment]
screen shot 2018-01-17 at 1 37 17 pm

**Comment by vivekmenezes** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx, line 126 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21411#-L2zjso1YKHJE9389Qfa:-L34PTjrYTbQh1m7mT8U:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b1d817d7ea3217e4b95592d2a76046ee5c47b04/pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx#L126)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

Should this be `units={AxisUnits.Duration}`?
</blockquote></details>

Done.

---

*[pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx, line 127 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21411#-L2cUm3-UZVd8xuDfoWz-r1-127:-L34PUlhjs5DQJNhSbee:b7094b2) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b1d817d7ea3217e4b95592d2a76046ee5c47b04/pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx#L127)):*
<details><summary><i>Previously, vilterp (Pete Vilter) wrote…</i></summary><blockquote>

looks like a typo in the metric name: it's `cr.node.clock-offset.meannanos`. https://github.com/cockroachdb/cockroach/blob/3dbe1dae8e734a27df53591f882efcd1543de9cd/pkg/rpc/clock_offset.go#L48
</blockquote></details>

Thanks!!!!!!

---

*[pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx, line 131 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21411#-L2zjumTs6KArPEMSrmA:-L34PWiK1sjb5X9ybiFR:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b1d817d7ea3217e4b95592d2a76046ee5c47b04/pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx#L131)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

Would it makes sense to combine these into a single chart?
</blockquote></details>

Done.

---

*[pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx, line 133 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/21411#-L2zjx15rOYrIhPBZ89x:-L34PXn0f_iTlYKMKZkW:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8b1d817d7ea3217e4b95592d2a76046ee5c47b04/pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/distributed.tsx#L133)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I would flip around the titles for both series, and spell this one out in full: "Clock Offset Mean" and "Clock Offset Standard Deviation"
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21411)*

**Comment by vivekmenezes** [comment]
I've address all the comments. Thanks!

**Comment by couchand** [comment]
:lgtm: Thanks!

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21411#-:-L34soHaHsxaYSPFoxMr:b-g2a5fb)*

**Pull request merged**